src/evdev/device.py

```diff
@@ -100,14 +102,14 @@ def __str__(self) -> str:
         return msg.format(*self)  # pylint: disable=not-an-iterable
 
 
-class InputDevice(EventIO):
+class InputDevice(EventIO, Generic[_AnyStr]):
     """
     A linux input device from which input events can be read.
     """
 
     __slots__ = ("path", "fd", "info", "name", "phys", "uniq", "_rawcapabilities", "version", "ff_effects_count")
 
-    def __init__(self, dev: Union[str, bytes, os.PathLike]):
+    def __init__(self, dev: Union[_AnyStr, os.PathLike[_AnyStr]]):
```

Comment:
Only needed for Python 3.8:
```suggestion
    def __init__(self, dev: Union[_AnyStr, "os.PathLike[_AnyStr]"]):
```

Reply:
I have never seen something like `os.PathLike[_AnyStr]` before, and even after reading https://stackoverflow.com/a/69680089 I don't understand what this does. Anyhow, apparently this is how it is done.

What do you think about calling the TypeVar ´PathType` instead of `_AnyStr`?

Reply:
I used the name from https://docs.python.org/3/library/typing.html#typing.AnyStr
As the deprecation notice says, it can be inlined later on when older Python releases are not supported, at which point the name is not relevant.

Reply:
> I have never seen something like `os.PathLike[_AnyStr]` before

You've never seen `list[str]` before? PathLike is no different.

As for TypeVars/Generic, I suggest looking at the mypy docs to understand all the basics:
https://mypy.readthedocs.io/en/stable/generics.html

Reply:
> `list[str]`

yeah I know that, and use it all the time. I have also used Generics before. Just not together with `os.PathLike`

Reply:
Right, well it's nothing special, just defines the precise type. Otherwise, it's always inferred to be `os.PathLike[Any]`, which then loses the return type for `__fspath__()`.

You can see the typing implementation in typeshed:
https://github.com/python/typeshed/blob/8064d995e716324473af6b5d415fbb8dd9b32316/stdlib/os/__init__.pyi#L878-L880

Reply:
Can we get this merged and released?

Reply:
Sorry, I have been extraordinarily busy

Reply:
All good. I've finally managed to remove the stubs I had in bthidhub, so it's all working now.